lecture_1/homework/app/controllers/samurais_controller.rb

```diff
@@ -0,0 +1,69 @@
+class SamuraisController < ApplicationController
+
+  def show
+    render json: samurai.to_json()
+  end
+
+  def index
+    limit = params.has_key?(:limit) ? params[:limit].to_i : 100
+    samurais = clan.samurais.first(limit)
+
+    # Check if shogun want's to display only alive/dead samurais and if so filter them
+    if params[:alive].to_s == "false"
+      samurais.select! { |samurai| samurai.died_at != nil } # TODO: Probably should also contain the check if it's a date
```

Comment:
Probably not :D tutaj tylko wyciągamy dane z bazy więc powinny być walidne :D Przy wprowadzeniu rzeczy do bazy można sprawdzać takie rzeczy :)

Reply:
Probably thank You ^^. Masz racje, często jak piszę zostawiam sobie takie todo'sy nawet jak nie jestem pewny co konkretnie mam na myśli, wychodzę z założenia, że lepiej trochę więcej razy przenalizować ten sam kod niż zostawić jakąś głupią dziurę, a nie zawsze się od razu ma czas/chęci w 100% poprawnie implementować.

Reply:
Hah, rozumiem :D